irods/webdav.yml

```
@@ -427,43 +427,61 @@
         path: /tmp/purgeman_setup
         state: absent
 
-    - name: create build dir for purgeman package 
-      file:
-        path: /tmp/purgeman_setup
-        state: directory
+    - name: check if purgeman is already installed
+      stat: 
+        path: /usr/bin/purgeman
+      register: purgemanResp
 
-    - name: download purgeman package from github and uncompress
-      unarchive:
-        src: https://github.com/cyverse/purgeman/releases/download/v0.3.0/purgeman_0.3.0.tar.gz
-        dest: /tmp/purgeman_setup
-        remote_src: True
-      register: unarchiveResp
+    - name: check purgeman version
+      shell: /usr/bin/purgeman -v -v | jq '.serviceVersion' | tr -d '"'
+      register: purgemanVersion
+      when: purgemanResp.stat.exists
 
-    - name: check if purgeman service exists
-      stat: 
-        path: /usr/lib/systemd/system/purgeman.service
-      register: serviceResp
-      tags:
-        - no_testing
+    - name: check purgeman service account
+      shell: grep purgeman -q /etc/passwd
+      register: purgemanAccount
+      ignore_errors: true
+      changed_when: false
 
-    - name: stop purgeman service if running
-      service:
-        name: purgeman
-        state: stopped
-      when: serviceResp.stat.exists
-      tags:
-        - no_testing
-     
-    - name: install purgeman service
-      when: unarchiveResp is changed
-      make: 
-        chdir: /tmp/purgeman_setup
-        target: install_centos
-
-    - name: delete build dir for purgeman package
-      file:
-        path: /tmp/purgeman_setup
-        state: absent
+    - when: (not purgemanResp.stat.exists) or (purgemanVersion.stdout != 'v0.3.0') or (purgemanAccount.rc != 0)
```

In ansible, using `ignore_errors` should be a last resort. It writes out a colorized error message to the terminal, which can mislead the person running ansible into thinking something failed, when it didn't. 

Instead of using `grep -q` to see if the account exists, you could use `id`. With `id`, instead of ignoring the error, you could use `failed_when: false`, because `id` writes an sensible error message to stderr, which can be checked to seeing if the account is missing like on line 446, e.g., `purgemanAccount.stderr != ''`.

---

Updated the code.